lib/p2p/NodeList.ts

```diff
@@ -0,0 +1,96 @@
+import P2PRepository from './P2PRepository';
+import { NodeFactory, NodeInstance } from '../types/db';
+
+/** Represents a list of nodes for managing network peers activity */
+class NodeList {
+  private nodes = new Map<string, NodeInstance>();
+
+  constructor(private repository: P2PRepository) {}
+
+  /**
+   * Return a [[NodeInstance]] array for nodes that haven't been banned.
+   */
+  public toArray = (): NodeInstance[] => {
+    const nodes: NodeInstance[] = Array.from({ length: this.nodes.size });
+    let unbannedNodeCount = 0;
+    this.nodes.forEach((node) => {
+      if (!node.banned) {
+        nodes[unbannedNodeCount] = node;
+        unbannedNodeCount += 1;
+      }
+    });
+    nodes.length = unbannedNodeCount;
+    return nodes;
+  }
+
+  /**
+   * Return a [[NodeFactory]] array for nodes that haven't been banned and have known addresses.
+   */
+  public toFactoryArray = (): NodeFactory[] => {
```

Comment:
I think using the `Factory` term here is a bit confusing, because it's from the DB row context. Maybe create a different type, with the same properties.

Reply:
That's true, so basically create an alias for the type? I'll look into it.

Reply:
Yep, something like this.